src/game/stage/stage.c

```diff
@@ -435,25 +477,46 @@ void HandleLifeLost(void)
 {
     gStageFlags |= STAGE_FLAG__DISABLE_PAUSE_MENU;
 
-    if (gGameMode == GAME_MODE_TIME_ATTACK || gGameMode == GAME_MODE_BOSS_TIME_ATTACK) {
+    if (GAME_MODE_IS_TIME_ATTACK) {
         TasksDestroyAll();
         PAUSE_BACKGROUNDS_QUEUE();
-        gUnknown_03005390 = 0;
+
+        SA2_LABEL(gUnknown_03005390) = 0;
+
         PAUSE_GRAPHICS_QUEUE();
         CreateTimeAttackLobbyScreen();
+#if (GAME == GAME_SA2)
         gNumLives = 2;
+#endif
         return;
-    }
-
-    if (--gNumLives == 0) {
-        gStageFlags |= STAGE_FLAG__ACT_START;
-        CreateGameOverScreen(OVER_CAUSE_ZERO_LIVES);
     } else {
-        TasksDestroyAll();
-        PAUSE_BACKGROUNDS_QUEUE();
-        gUnknown_03005390 = 0;
-        PAUSE_GRAPHICS_QUEUE();
-        CreateGameStage();
+#if (GAME == GAME_SA1)
+        if (IS_SINGLE_PLAYER && (--gNumLives == 0))
```

Comment:
Could make this simpler by adding a `if (IS_MULTIPLAYER) { return; }` and then wrapping that in the sa1 macro?

Reply:
Doesn't seem to match :/

Reply:
Which makes sense because of the else block

Reply:
Ah true